### PR TITLE
Use ioredis package for routerlicious server 

### PR DESCRIPTION
**Pull request by TriDucNgo**
This change updates the packages under **/server/routerlicious/...** to begin using the [ioredis ](https://github.com/luin/ioredis)package over [node-redis](https://github.com/NodeRedis/node-redis). Because this change is not backwards-compatible (due to different type signatures of the 2 libraries), Other areas of the server code outside of **/server/routerlicious/...** such as [Historian](https://github.com/microsoft/FluidFramework/blob/b992327c1592b4ca3add78881cffb595c65439ec/server/historian/packages/historian-base/src/runnerFactory.ts#L44) and [Gateway](https://github.com/microsoft/FluidFramework/blob/b992327c1592b4ca3add78881cffb595c65439ec/server/gateway/src/runnerFactory.ts#L63) that uses Redis will continue to use node-redis until new versions of the routerlicious packages are available with these ioredis changes, after which they can be updated to use ioredis as well.

### CODE REVIEW

File: server/routerlicious/packages/services-shared/src/socketIoRedisConnection.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class SocketIoRedisSubscriptionConnection
      */
     private readonly subscriptions: Map<string, (channel: string, messageBuffer: Buffer) => void> = new Map();
 
-    constructor(client: redis.RedisClient) {
+    constructor(client: Redis) {
         super(client);
```
Comment (hedasilv) [inline server/routerlicious/packages/services-shared/src/socketIoRedisConnection.ts:35]: Do we want to add an error listener for the Redis client such as in server/routerlicious/packages/services/src/redisClientManager.ts?
```ts
client.on("error", (error) => {
            winston.error("Client Manager Redis Error:", error);
        });
```

File: server/routerlicious/packages/services/src/redis.ts [hunk 2]
```diff
@@ -3,20 +3,18 @@
...
 export class RedisCache implements ICache {
     private readonly getAsync: any;
     private readonly setAsync: any;
 
-    constructor(client: RedisClient, private readonly prefix = "page") {
-        this.getAsync = util.promisify(client.get.bind(client));
-        this.setAsync = util.promisify(client.set.bind(client));
+    constructor(client: Redis, private readonly prefix = "page") {
+        this.getAsync = client.get.bind(client);
+        this.setAsync = client.set.bind(client);
```
Comment (hedasilv) [inline server/routerlicious/packages/services/src/redis.ts:17]: Do we want to add an error listener for the Redis client such as in server/routerlicious/packages/services/src/redisClientManager.ts?
```ts
client.on("error", (error) => {
            winston.error("Client Manager Redis Error:", error);
        });
```

**Review by hedasilv — APPROVED** [review]

File: server/routerlicious/packages/services/src/redisClientManager.ts [hunk 3]
```diff
@@ -16,13 +16,17 @@ export class ClientManager implements IClientManager {
...
         private readonly expireAfterSeconds = 60 * 60 * 24,
         private readonly prefix = "client") {
-        this.addAsync = util.promisify(client.hmset.bind(client));
-        this.removeAsync = util.promisify(client.hdel.bind(client));
-        this.findAllAsync = util.promisify(client.hgetall.bind(client));
-        this.expire = util.promisify(client.expire.bind(client));
+        this.addAsync = client.hmset.bind(client);
+        this.removeAsync = client.hdel.bind(client);
+        this.findAllAsync = client.hgetall.bind(client);
+        this.expire = client.expire.bind(client);
```
Comment (tanviraumi) [inline server/routerlicious/packages/services/src/redisClientManager.ts:25]: If these new libraries are already promisified, can we just use those directly?

File: server/routerlicious/packages/services/src/redisThrottleStorageManager.ts [hunk 4]
```diff
@@ -19,13 +19,17 @@ export class RedisThrottleStorageManager implements IThrottleStorageManager {
...
+        client: Redis,
         private readonly expireAfterSeconds = 60 * 60 * 24,
         private readonly prefix = "throttle",
     ) {
-        this.setAsync = util.promisify(client.hmset.bind(client));
-        this.getAsync = util.promisify(client.hgetall.bind(client));
-        this.expire = util.promisify(client.expire.bind(client));
+        this.setAsync = client.hmset.bind(client);
+        this.getAsync = client.hgetall.bind(client);
+        this.expire = client.expire.bind(client);
```
Comment (tanviraumi) [inline server/routerlicious/packages/services/src/redisThrottleStorageManager.ts:28]: Same as above

**Review by hedasilv — APPROVED** [review]

**Review by tanviraumi — APPROVED** [review]

**Comment by tanviraumi** [comment]
@GaryWilber Now that we are moving to ioredis, should we able to use one single logic? (https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts#L124)

**Comment by GaryWilber** [comment]
> @GaryWilber Now that we are moving to ioredis, should we able to use one single logic? (https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts#L124)

Yea that would be nice

File: server/routerlicious/packages/services/src/redisClientManager.ts (hunk 3 above)
Comment (TriDucNgo) [reply server/routerlicious/packages/services/src/redisClientManager.ts]: I went ahead and removed the class' custom fields and used the redis client directly.

**Review by hedasilv — APPROVED** [review]

**Review by tanviraumi — APPROVED** [review]

**Review by znewton — APPROVED** [review]

**Pull request merged**